packages/swapper/src/swappers/zrx/ZrxApprovalNeeded/ZrxApprovalNeeded.ts

```diff
@@ -13,36 +13,54 @@ export async function ZrxApprovalNeeded(
 ): Promise<ApprovalNeededOutput> {
   const { sellAsset } = quote
 
-  if (sellAsset.chainId !== 'eip155:1') {
-    throw new SwapError('ZrxSwapper:ZrxApprovalNeeded only Ethereum chain type is supported')
-  }
+  try {
+    if (sellAsset.chainId !== 'eip155:1') {
+      throw new SwapError('[ZrxApprovalNeeded] - sellAsset chainId is not supported', {
+        code: SwapErrorTypes.APPROVAL_NEEDED,
+        details: { chainId: sellAsset.chainId }
+      })
+    }
 
-  if (sellAsset.symbol === 'ETH') {
-    return { approvalNeeded: false }
-  }
+    // No approval needed for trading eth itself
+    if (sellAsset.assetId === 'eip155:1/slip44:60') {
+      return { approvalNeeded: false }
+    }
 
-  const accountNumber = quote.sellAssetAccountId ? Number(quote.sellAssetAccountId) : 0
+    const accountNumber = bnOrZero(quote.sellAssetAccountId).toNumber()
 
-  const adapter = await adapterManager.byChainId(sellAsset.chainId)
-  const bip44Params = adapter.buildBIP44Params({ accountNumber })
-  const receiveAddress = await adapter.getAddress({ wallet, bip44Params })
+    const adapter = await adapterManager.byChainId(sellAsset.chainId)
+    const bip44Params = adapter.buildBIP44Params({ accountNumber })
+    const receiveAddress = await adapter.getAddress({ wallet, bip44Params })
 
-  if (!quote.sellAsset.tokenId || !quote.allowanceContract) {
-    throw new SwapError('ZrxApprovalNeeded - tokenId and allowanceTarget are required')
-  }
+    if (!quote.sellAsset.tokenId || !quote.allowanceContract) {
```

Comment:
i don't want to see `tokenId` in new code - this should actually be removed from the asset type. all the information is encoded in the assetId - do this instead

```
const { assetReference } = fromAssetId(quote.sellAsset.assetId) // per CAIP19, assetReference *is* the tokenId
```

Reply:
https://github.com/shapeshift/lib/issues/652